alarm_active_table.cpp

```diff
@@ -0,0 +1,423 @@
+                                 unsigned long index,
+                                 AlarmTableDef& def)
+{
+  alarmActiveTable_context* ctx = SNMP_MALLOC_TYPEDEF(alarmActiveTable_context);
+  ctx->_alarm_table_def = &def;
+  if (pointers[ctx->_alarm_table_def->alarm_index()])
+  {
+    if (ctx->_alarm_table_def->severity() == pointers[ctx->_alarm_table_def->alarm_index()]->_alarm_table_def->severity())
+    {
+      return;
+    }
+  }
+  if (!ctx)
+  {
+    TRC_ERROR("malloc failed: alarmActiveTable_create_row");
+    return;
+  }
+        
+  if (alarmActiveTable_index_to_oid(name, datetime, index, &ctx->_index) != SNMP_ERR_NOERROR)
+  {
+    if (ctx->_index.oids != NULL)
+    {
+      free(ctx->_index.oids);
+    }
+
+    free(ctx);
+    return;
+  }
+  if (ctx->_alarm_table_def->severity() == 1)
+  {
+    alarmActiveTable_delete_row(def);
+  }
+  else
+  {
+    if (pointers[ctx->_alarm_table_def->alarm_index()])
+    {
+      CONTAINER_REMOVE(cb.container, pointers[ctx->_alarm_table_def->alarm_index()]);
+    }
+    
+    if (ctx)
+    {
+      CONTAINER_INSERT(cb.container, ctx);
+      pointers[ctx->_alarm_table_def->alarm_index()] = ctx;
+    }
+  }
+  return;
+}
+
+/************************************************************
+ *
+ * Delete a row
+ */
+void alarmActiveTable_delete_row(AlarmTableDef& def)
+{
+  alarmActiveTable_context* ctx = SNMP_MALLOC_TYPEDEF(alarmActiveTable_context);
+  ctx->_alarm_table_def = &def;
+  if (!ctx)
+  {
+    TRC_ERROR("malloc failed: alarmActiveTable_remove_row");
+    return;
+  }
+  
+  if (ctx)
+  {
+    CONTAINER_REMOVE(cb.container, pointers[ctx->_alarm_table_def->alarm_index()]);
+    pointers[ctx->_alarm_table_def->alarm_index()] = NULL;
+  }
+}
+
+/************************************************************
+ *
+ *  Convert table index components to an oid.
+ */
+int alarmActiveTable_index_to_oid(char* name,
+                                 alarmActiveTable_SNMPDateTime* datetime,
+                                 unsigned long counter,
+                                 netsnmp_index *oid_idx)
+{
+  int err = SNMP_ERR_NOERROR;
+
+
+  netsnmp_variable_list var_alarmListName;
+  netsnmp_variable_list var_alarmActiveDateAndTime;
+  netsnmp_variable_list var_alarmActiveIndex;
+  /*
+   * set up varbinds
+   */
+  memset(&var_alarmListName, 0x00, sizeof(var_alarmListName));
+  var_alarmListName.type = ASN_OCTET_STR;
+  memset(&var_alarmActiveDateAndTime, 0x00, sizeof(var_alarmActiveDateAndTime));
+  var_alarmActiveDateAndTime.type = ASN_OCTET_STR;
+  memset(&var_alarmActiveIndex, 0x00, sizeof(var_alarmActiveIndex));
+  var_alarmActiveIndex.type = ASN_UNSIGNED;
+
+  /*
+   * chain index varbinds together
+   */
+  var_alarmListName.next_variable = &var_alarmActiveDateAndTime;
+  var_alarmActiveDateAndTime.next_variable = &var_alarmActiveIndex;
+  var_alarmActiveIndex.next_variable =  NULL;
+
+
+  DEBUGMSGTL(("verbose:alarmActiveTable:alarmActiveTable_index_to_oid", "called\n"));
```

Comment:
I don't think this is useful. I think this is just telling us the function alarmActiveTable_index_to_oid has been called so I suggest removing this line altogether.